bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/MetadataCommandDescriptionProvider.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.internal.items;
+
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.internal.types.CommandDescriptionImpl;
+import org.openhab.core.items.Metadata;
+import org.openhab.core.items.MetadataKey;
+import org.openhab.core.items.MetadataRegistry;
+import org.openhab.core.types.CommandDescription;
+import org.openhab.core.types.CommandDescriptionProvider;
+import org.openhab.core.types.CommandOption;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link CommandDescription} provider from items' metadata
+ *
+ * @author Yannick Schaus - initial contribution
+ *
+ */
+@Component(service = CommandDescriptionProvider.class)
+public class MetadataCommandDescriptionProvider implements CommandDescriptionProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(MetadataCommandDescriptionProvider.class);
+
+    public static final String COMMANDDESCRIPTION_METADATA_NAMESPACE = "commandDescription";
+
+    private MetadataRegistry metadataRegistry;
+
+    @Override
+    public @Nullable CommandDescription getCommandDescription(@NonNull String itemName, @Nullable Locale locale) {
+        Metadata metadata = metadataRegistry.get(new MetadataKey(COMMANDDESCRIPTION_METADATA_NAMESPACE, itemName));
+
+        if (metadata != null) {
+            try {
+                CommandDescriptionImpl commandDescription = new CommandDescriptionImpl();
+                if (metadata.getConfiguration().containsKey("options")) {
+                    Stream.of(metadata.getConfiguration().get("options").toString().split(",")).forEach(o -> {
+                        if (o.contains("=")) {
+                            commandDescription.addCommandOption(
+                                    new CommandOption(o.split("=")[0].trim(), o.split("=")[1].trim()));
+                        } else {
+                            commandDescription.addCommandOption(new CommandOption(o.trim(), null));
+                        }
+                    });
+                }
+
+                return commandDescription;
+            } catch (Exception e) {
+                logger.error("Unable to parse the commandDescription from metadata for item {}", itemName, e);
+                return null;
+            }
+        }
+
+        return null;
+    }
+
+    @Reference
```

Comment:
You can add the `@Reference` directly in line 45 and get rid off the getter and setter methods.
Or pass it in via constructor as you do in `MetadataStateDescriptionFragmentProvider`.